streaming/base/format/base/writer.py

```diff
@@ -116,6 +120,13 @@ def __init__(self,
 
         self.shards = []
 
+        # Remove local directory if requested prior to creating writer
+        local = out if isinstance(out, str) else out[0]
+        if os.path.exists(local) and kwargs.get('exist_ok', False):
```

Comment:
shouldn't this be kwargs.get('exist_ok', True)?

Reply:
I would say the directory should not be removed by default - files should not be deleted unless the user has explicitly set `exist_ok` to True.

Reply:
My understanding is:
- If `local` exists but `exist_ok` is False (default value), then `local` will not be removed and `CloudUploader` will throw an error. Avoids accidentally overwriting data.
- If `local` doesn't exist then nothing needs deleting and `CloudUploader` will work fine.
- If `local` exists and `exist_ok` is True (has to be set by the user), then `local` will be removed with a warning and `CloudUploader` will work fine.